Meta/CMake/Swift/swift-settings.cmake

```diff
@@ -60,8 +60,13 @@ function(add_swift_target_properties target_name)
     get_target_property(_NATIVE_DIRS ${target_name} INCLUDE_DIRECTORIES)
     list(APPEND _NATIVE_DIRS ${CMAKE_Swift_MODULE_DIRECTORY})
 
+    set(EXTRA_COMPILE_DEFINITIONS "")
+    foreach (compile_definition IN LISTS SWIFT_TARGET_COMPILE_DEFINITIONS)
+        list(APPEND EXTRA_COMPILE_DEFINITIONS "-Xcc" "-D${compile_definition}")
+    endforeach()
+
     _swift_generate_cxx_header(${target_name} "${target_name}-Swift.h"
         SEARCH_PATHS ${_NATIVE_DIRS}
-        COMPILE_OPTIONS ${VFS_OVERLAY_OPTIONS}
+        COMPILE_OPTIONS ${VFS_OVERLAY_OPTIONS} ${EXTRA_COMPILE_DEFINITIONS} ${SWIFT_TARGET_COMPILE_OPTIONS}
```

Comment:
@etcwilde I was wondering if there were any updates on getting this -emit-clang-header-path feature thingy into CMake itself? As the build here gets more complex, I keep finding that I have to repeat information that CMake already "knows" about my target:

- imported compile definitions (-DMY_WORKAROUND_FLAG)
- imported compile options (swift-testing plugin path)
- global swift compile options (-enable-experimental-feature Extern)